src/test/java/com/example/exercise2/controller/BookControllerTest.java

```diff
@@ -0,0 +1,121 @@
+package com.example.exercise2.controller;
+
+import static org.hamcrest.core.StringContains.containsString;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.example.exercise2.service.BookService;
+import com.example.exercise2.service.model.Book;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+@WebMvcTest(BookController.class)
+@AutoConfigureMockMvc
+public class BookControllerTest {
+
+  @Autowired
+  private MockMvc mockMvc;
+
+  @MockBean
+  private BookService bookService;
+
+  private ObjectMapper objectMapper = new ObjectMapper();
+
+  @Test
+  @DisplayName("Given a successful save, response should give http status 200")
+  void success() throws Exception {
+
+    //setup
+    Book bookRequest = new Book();
+    bookRequest.setId(1L);
+    bookRequest.setAuthor("Mr Z");
+    bookRequest.setTitle("Science 3");
+
+    Book expectedResponse = new Book();
+    expectedResponse.setId(1L);
+    expectedResponse.setAuthor("Mr Z");
+    expectedResponse.setTitle("Science 3");
+
+    when(bookService.save(any())).thenReturn(expectedResponse);
+
```

Comment:
I know it is very tempting to make a stub work by using any matcher, buyt as much as possible, we should avoid it, having this means there is nothing enforcing someone to hardcode an input of any value in ytour book service.